src/chage.c

```diff
@@ -243,7 +243,7 @@ print_day_as_date(long day)
 	}
 
 	if (localtime_r(&date, &tm) == NULL) {
-		(void) printf ("time_t: %lu\n", (unsigned long)date);
+		puts(_("future"));
```

Comment:
Why did you change `printf()`?

Reply:
```
commit e78f9782183e88e3973d4634d22c538a09f44b96
Author: Alejandro Colomar <alx@kernel.org>
Date:   Wed Jul 31 14:53:12 2024 +0200

    src/chage.c: print_day_as_date(): Simplify error handling
    
    If localtime_r(3) fails, just print future, as we do in day_to_str().
    It should only fail for unrealistic dates, if at all.
    
    This is a step towards reusing the DAY_TO_STR() function.
    
    Signed-off-by: Alejandro Colomar <alx@kernel.org>

diff --git a/src/chage.c b/src/chage.c
index 0f963090..8b2d7b35 100644
--- a/src/chage.c
+++ b/src/chage.c
@@ -243,7 +243,7 @@ print_day_as_date(long day)
        }
 
        if (localtime_r(&date, &tm) == NULL) {
-               (void) printf ("time_t: %lu\n", (unsigned long)date);
+               puts(_("future"));
                return;
        }
 
```

This makes it behave like str_to_day(), so that we can just call it there.

Reply:
```diff
$ diff -u <(git show shadow/master:src/chage.c | grepc print_day_as_date) <(git show shadow/master:lib/time/day_to_str.h | grepc day_to_str)
--- /dev/fd/63	2024-08-02 11:49:56.228997826 +0200
+++ /dev/fd/62	2024-08-02 11:49:56.228997826 +0200
@@ -1,25 +1,25 @@
-(standard input):static void print_day_as_date (long day);
-(standard input):static void
-print_day_as_date(long day)
+(standard input):inline void day_to_str(size_t size, char buf[size], long day);
+(standard input):inline void
+day_to_str(size_t size, char buf[size], long day)
 {
-	char       buf[80];
 	time_t     date;
 	struct tm  tm;
 
 	if (day < 0) {
-		puts(_("never"));
+		strtcpy(buf, "never", size);
 		return;
 	}
+
 	if (__builtin_mul_overflow(day, DAY, &date)) {
-		puts(_("future"));
+		strtcpy(buf, "future", size);
 		return;
 	}
 
 	if (localtime_r(&date, &tm) == NULL) {
-		(void) printf ("time_t: %lu\n", (unsigned long)date);
+		strtcpy(buf, "future", size);
 		return;
 	}
 
-	STRFTIME(buf, iflg ? "%Y-%m-%d" : "%b %d, %Y", &tm);
-	(void) puts (buf);
+	if (strftime(buf, size, "%Y-%m-%d", &tm) == 0)
+		strtcpy(buf, "future", size);
 }
```

Reply:
So, after that commit, the difference is reduced to just:

```diff
$ diff -u \
    <(git show e78f9782183e:src/chage.c \
      | grepc print_day_as_date) \
    <(git show e78f9782183e:lib/time/day_to_str.h \
      | grepc day_to_str);
--- /dev/fd/63	2024-08-02 14:14:36.069372059 +0200
+++ /dev/fd/62	2024-08-02 14:14:36.069372059 +0200
@@ -1,25 +1,25 @@
-(standard input):static void print_day_as_date (long day);
-(standard input):static void
-print_day_as_date(long day)
+(standard input):inline void day_to_str(size_t size, char buf[size], long day);
+(standard input):inline void
+day_to_str(size_t size, char buf[size], long day)
 {
-	char       buf[80];
 	time_t     date;
 	struct tm  tm;
 
 	if (day < 0) {
-		puts(_("never"));
+		strtcpy(buf, "never", size);
 		return;
 	}
+
 	if (__builtin_mul_overflow(day, DAY, &date)) {
-		puts(_("future"));
+		strtcpy(buf, "future", size);
 		return;
 	}
 
 	if (localtime_r(&date, &tm) == NULL) {
-		puts(_("future"));
+		strtcpy(buf, "future", size);
 		return;
 	}
 
-	STRFTIME(buf, iflg ? "%F" : "%x", &tm);
-	(void) puts (buf);
+	if (strftime(buf, size, "%F", &tm) == 0)
+		strtcpy(buf, "future", size);
 }
```

So the only difference after that is the support for localized dates.